app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -572,8 +598,8 @@ public CustomProgressDialog getCurrentProgressDialog() {
     @Override
     public void dismissProgressDialog() {
         CustomProgressDialog progressDialog = getCurrentProgressDialog();
-        if (progressDialog != null && progressDialog.isAdded()) {
-            progressDialog.dismissAllowingStateLoss();
+        if (!areFragmentsPaused && progressDialog != null && progressDialog.isAdded()) {
+            progressDialog.dismiss();

```

Comment:
Forces us to be more careful with fragment transactions. We should always try to avoid `*AllowingStateLoss` because it is prone to introduce ui bugs.